core/src/test/java/org/apache/calcite/test/ModelTest.java

```diff
@@ -342,12 +344,39 @@ private ObjectMapper mapper() {
     final JsonView table1 = (JsonView) schema.tables.get(0);
     try {
       String s = table1.getSql();
-      fail("exprcted error, got " + s);
+      fail("expected error, got " + s);
     } catch (RuntimeException e) {
       assertThat(e.getMessage(),
           equalTo("each element of a string list must be a string; found: 2"));
     }
   }
+
+  @Test public void testYamlInlineDetection() throws Exception {
+    // yaml model with different line endings
+    final String yamlModel = "version: 1.0\r\n"
+        + "schemas: \n"
+        + "- type: custom\r\n"
+        + "  name: 'MyCustomSchema'\n"
+        + "  factory: " + JdbcTest.MySchemaFactory.class.getName() + "\r\n";
+    CalciteAssert.model(yamlModel).doWithConnection(calciteConnection -> null);
+    // with a comment
+    CalciteAssert.model("\n  \r\n# comment\n " + yamlModel)
+        .doWithConnection(calciteConnection -> null);
+    // if starts with { => treated as json
+    CalciteAssert.model("  { " + yamlModel + " }")
+        .connectThrows("Unexpected character ('s' (code 115)): "
+            + "was expecting comma to separate Object entries");
+    // if starts with /* => treated as json
+    CalciteAssert.model("  /* " + yamlModel)
+        .connectThrows("Unexpected end-of-input in a comment");
+  }
+
+  @Test public void testYamlFileDetection() throws Exception {
+    final URL inUrl = ModelTest.class.getResource("/empty-model.yaml");
+    CalciteAssert.that()
+        .with(CalciteConnectionProperty.MODEL, inUrl.getFile())
```

Comment:
Apparently this fails: https://builds.apache.org/job/Calcite-Master/805/jdk=JDK%201.8%20(latest),label_exp=ubuntu&&!cloud-slave/console
It should be `withModel(inUrl)`